src/polus/plugins/_plugins/classes/plugin_classes.py

```diff
@@ -97,18 +101,34 @@ def refresh():
             if file.suffix == ".py":
                 continue
 
-            plugin = validate_manifest(file)
-            key = name_cleaner(plugin.name)
-            # Add version and path to VERSIONS
-            if key not in PLUGINS:
-                PLUGINS[key] = {}
-            if plugin.version in PLUGINS[key]:
-                if not file == PLUGINS[key][plugin.version]:
-                    raise DuplicateVersionFound(
-                        "Found duplicate version of plugin %s in %s"
-                        % (plugin.name, PLUGIN_DIR)
-                    )
-            PLUGINS[key][plugin.version] = file
+            try:
+                plugin = validate_manifest(file)
+            except InvalidManifest as e:
+                _invalid[_load_manifest(file)["name"]] = str(e.__cause__)
```

Comment:
Seem to be a problem in the logic here. If we the manifest do not contain a name, validate_manifest will throw an error but _load_manifest(file)["name"] won't exist.
I suggest creating some tests to demonstrate how errors are handled in different cases

Reply:
Removed this, now a warning including the manifest path gets logged when `InvalidManifest` is raised